### PR TITLE
Simplify js tester a bit

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @notriddle

### CODE REVIEW

File: src/tools/rustdoc-js/tester.js [hunk 1]
```diff
@@ -423,11 +423,7 @@ function parseOptions(args) {
     };
 
     for (var i = 0; i < args.length; ++i) {
-        if (args[i] === "--resource-suffix"
-            || args[i] === "--doc-folder"
-            || args[i] === "--test-folder"
-            || args[i] === "--test-file"
-            || args[i] === "--crate-name") {
+        if (correspondances[args[i]] !== undefined) {
```
Comment (notriddle) [inline src/tools/rustdoc-js/tester.js:426]:
```suggestion
        if (Object.hasOwn(correspondances, args[i])) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn

Comment (camelid) [reply src/tools/rustdoc-js/tester.js:426]: Also, a nit: "correspondances" should be spelled "correspondences". While you're at it, do you want to fix the spelling?

Comment (GuillaumeGomez) [reply src/tools/rustdoc-js/tester.js:426]: Sure!

**Comment by GuillaumeGomez** [comment]
Updated. TIL thanks to both of you. :)

**Comment by GuillaumeGomez** [comment]
@notriddle I didn't check the requirements but they're quite high in fact! I'll revert to `hasOwnProperty`.

**Comment by GuillaumeGomez** [comment]
So apparently, `Object.hasOwnProperty` isn't a thing, but `obj.hasOwnProperty` is. Oh well, JS.

**Review by notriddle — APPROVED** [review]

**Pull request merged**